lib/Raven/Client.php

```diff
@@ -747,233 +727,40 @@ public function send(&$data)
-     * @param array        $headers Associative array of headers
-     */
-    protected function send_http($url, $data, $headers = [])
-    {
-        if ($this->config->getCurlMethod() == 'async') {
-            $this->_curl_handler->enqueue($url, $data, $headers);
-        } elseif ($this->config->getCurlMethod() == 'exec') {
-            $this->send_http_asynchronous_curl_exec($url, $data, $headers);
-        } else {
-            $this->send_http_synchronous($url, $data, $headers);
-        }
-    }
-
-    /**
-     * @param string $url
-     * @param string $data
-     * @param array  $headers
-     * @return string
-     *
-     * This command line ensures exec returns immediately while curl runs in the background
-     */
-    protected function buildCurlCommand($url, $data, $headers)
-    {
-        $post_fields = '';
-        foreach ($headers as $key => $value) {
-            $post_fields .= ' -H '.escapeshellarg($key.': '.$value);
-        }
-        $cmd = sprintf(
-            '%s -X POST%s -d %s %s -m %d %s%s> /dev/null 2>&1 &',
-            escapeshellcmd($this->config->getCurlPath()), $post_fields,
-            escapeshellarg($data), escapeshellarg($url), $this->config->getTimeout(),
-            !$this->config->isSslVerificationEnabled() ? '-k ' : '',
-            !empty($this->config->getSslCaFile()) ? '--cacert '.escapeshellarg($this->config->getSslCaFile()).' ' : ''
+        $request = $this->requestFactory->createRequest(
+            'POST',
+            sprintf('api/%d/store/', $this->getConfig()->getProjectId()),
+            ['Content-Type' => 'gzip' === $this->config->getEncoding() ? 'application/octet-stream' : 'application/json'],
+            JSON::encode($data)
         );
 
-        return $cmd;
-    }
-
-    /**
-     * Send the cURL to Sentry asynchronously. No errors will be returned from cURL
-     *
-     * @param string       $url     URL of the Sentry instance to log to
-     * @param array|string $data    Associative array of data to log
-     * @param array        $headers Associative array of headers
-     * @return bool
-     */
-    protected function send_http_asynchronous_curl_exec($url, $data, $headers)
-    {
-        exec($this->buildCurlCommand($url, $data, $headers));
-        return true; // The exec method is just fire and forget, so just assume it always works
-    }
-
-    /**
-     * Send a blocking cURL to Sentry and check for errors from cURL
-     *
-     * @param string       $url     URL of the Sentry instance to log to
-     * @param array|string $data    Associative array of data to log
-     * @param array        $headers Associative array of headers
-     * @return bool
-     */
-    protected function send_http_synchronous($url, $data, $headers)
-    {
-        $new_headers = [];
-        foreach ($headers as $key => $value) {
-            array_push($new_headers, $key .': '. $value);
+        if ('gzip' === $this->config->getEncoding()) {
```

Comment:
Can we translate this in a `isEncodingCompresed` method? I would like to avoid having the `gzip` string hardcoded and written everywhere.